shared/src/uniswap_pool.rs

```diff
@@ -69,25 +58,35 @@ impl Pool {
     /// Given an output amount and token, returns a required input amount and address of the other asset.
     /// Returns None if operation not possible due to arithmetic issues (e.g. over or underflow, reserve too small)
     pub fn get_amount_in(&self, token_out: H160, amount_out: U256) -> Option<(U256, H160)> {
+        let (reserve_out, reserve_in, token_in) = self.get_relative_reserves(token_out);
```

Comment:
I'm personally not a huge fan of splitting this out into its own method, because it disconnects the logic from the ordering of the tuple. If we want the extra method, I'd slightly prefer a named struct as a return type to avoid mixup of `reserve_in`, `reserve_out` (both are U256). I can see how this is a lot of boilerplate code for arguably little benefit as all of this is implementation detail to this component and likely not used outside. 

But then my general question is: What is the benefit of the extra method. Anyways, super nitty and I'm fine either.

Reply:
I consider it a valuable method that could potentially be used for another purpose at some point. Not generally a fan of performing swaps like this in a method that should be easy follow. In my opinion such swaps should always be delegated to their own helper methods. Happy to restructure if you'd like but I find the readability of the amount in and amount out methods now much simpler to follow.